update docs for Node.js (#21)
signalfx/splunk-otel-lambda: README.md

The following components are currently supported:
- [Java wrapper](#java-wrapper) 
- [Python wrapper](#python-wrapper)
- [Node.js wrapper](#node.js-wrapper)

### Configuration

> All components are preconfigured to use Splunk as the tracing backend.

Reviewer: ```suggestion
All components are preconfigured to use Splunk Observability Cloud as the tracing backend.
```

Author: done

   
2. Traces export

    By default, all wrappers use the OTLP/gRPC exporter. 

>     If the `SPLUNK_REALM` environment variable is set, the value of  the`OTEL_EXPORTER_OTLP_ENDPOINT` environment variable is `https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace/otlp`, the direct ingest URL for Splunk Observability Cloud. 

Reviewer 2: Nit
```suggestion
    If the `SPLUNK_REALM` environment variable is set, the value of the`OTEL_EXPORTER_OTLP_ENDPOINT` environment variable is `https://ingest.${SPLUNK_REALM}.signalfx.com/v2/trace/otlp`, the direct ingest URL for Splunk Observability Cloud. 
```

    ```
    OTEL_EXPORTER_OTLP_ENDPOINT: <endpoint URL>
    ```

>     

Reviewer 2: Dangling whitespace
```suggestion

```

    OTEL_EXPORTER_JAEGER_ENDPOINT: <endpoint URL>
    ```
   
   > **_NOTE:_**  Currently Splunk Observability Cloud supports `jaeger/thrift` and `OTLP` with `http/protobuf` protocols. In order to send traces directly to ingest, set following environment variable in your Lambda function code:
   > - Java/Python - `OTEL_EXPORTER_OTLP_TRACES_PROTOCOL: http/protobuf`

>    > - Node.js - `OTEL_TRACES_EXPORTER: jaeger-thrift-splunk`

Author: Well, because Node.js does not support http/protobuf yet! ;)

    OTEL_EXPORTER_JAEGER_ENDPOINT: <endpoint URL>
    ```
   

>    > **_NOTE:_**  Currently Splunk Observability Cloud supports `jaeger/thrift` and `OTLP` with `http/protobuf` protocols. In order to send traces directly to ingest, set following environment variable in your Lambda function code:

Reviewer 2: I think it'd probably be better to use `Jaeger Thrift` -- `jaeger-thrift` looks like it could be confused with the OTLP protocol.

    For more information about available context propagators, see the [Propagator settings](https://github.com/signalfx/splunk-otel-python/blob/main/docs/advanced-config.md#trace-propagation-configuration) for the Splunk distribution of OpenTelemetry Python.

### Node.js wrapper

> The Node.js wrapper is based on Splunk Distribution of OpenTelemetry for Node.js version `0.14.0`. 

Author: https://github.com/open-telemetry/opentelemetry-js#supported-runtimes - that's why I wanted to leave base distro docs link :)

> The Node.js wrapper is based on Splunk Distribution of OpenTelemetry for Node.js version `0.14.0`. 

Reviewer: We're not supporting node 16 though?

> The Node.js wrapper is based on Splunk Distribution of OpenTelemetry for Node.js version `0.14.0`. 

Author: I believe we support everything that upstream does. @jtmal-signalfx any word on this?

> The official documentation of the base distribution can be found [here](https://github.com/signalfx/splunk-otel-js).

Reviewer: ```suggestion
The official documentation of the base distribution can be found [here](https://docs.splunk.com/Observability/gdi/get-data-in/application/nodejs/get-started.html
).
```

Author: I'd leave tech docs here - they contain advanced conf, compat and other info.

> The official documentation of the base distribution can be found [here](https://github.com/signalfx/splunk-otel-js).

Reviewer: We must call those docs something else then—they're not the "official" Splunk docs. How about just "documentation"?

```suggestion
The documentation of the base distribution can be found [here](https://github.com/signalfx/splunk-otel-js).
```

Author: done

    For more information about available context propagators, see the [Propagator settings](https://github.com/signalfx/splunk-otel-python/blob/main/docs/advanced-config.md#trace-propagation-configuration) for the Splunk distribution of OpenTelemetry Python.

### Node.js wrapper

> The Node.js wrapper is based on Splunk Distribution of OpenTelemetry for Node.js version `0.14.0`. 

Reviewer 2: Are you sure you want to hard-code versions? 😄

Author: Yep, each version change of dependencies will require a PR anyway.

AWS_LAMBDA_EXEC_WRAPPER: /opt/otel-instrument
SPLUNK_ACCESS_TOKEN: <org_access_token>
SPLUNK_REALM: us0
OTEL_EXPORTER_OTLP_TRACES_PROTOCOL: http/protobuf
```

#### Node.js

The example assumes the following:

- Data is sent to Splunk APM directly via Ingest endpoint.
- Context propagation is default (`tracecontext,baggage`).
- Realm is `us0`.

Following environment variables should be set:
```
AWS_LAMBDA_EXEC_WRAPPER: /opt/nodejs-otel-handler
SPLUNK_ACCESS_TOKEN: <org_access_token>
SPLUNK_REALM: us0
OTEL_TRACES_EXPORTER: jaeger-thrift-splunk
```

## Troubleshooting

### Java

1. If traces are not showing in APM 
    - Set `OTEL_LOG_LEVEL` to `DEBUG` and check the CloudWatch logs.
    - Set `OTEL_LAMBDA_LOG_LEVEL` to `DEBUG` and search for traces IDs in the backend.

>     - Set `OTEL_LAMBDA_LOG_LEVEL` to `DEBUG` for the `jaeger-thrift-splunk' exporter. This shows debug information for the exporter.

Reviewer 2: How's that different from the previous step?

Author: With j/t exporter you will see additional info - ie actual calls to the backend. So it's about the combination rather than a single value.
